### PR TITLE
feat: add load more button to grant beneficiary box

### LINKED ISSUE #1381: No pagination in Grant beneficiary widget on profile view
See: https://governance.decentraland.org/profile/?address=0x511a22cdd2c4ee8357bb02df2578037ffe8a4d8d
Only 4 Grants appear on the widget when in reality this profile asked for 5 grants.

Definition: Let's show 3 Grants and then the Load more button.

### PR DESCRIPTION
**Pull request by andyesp**
**Screenshots**
image

**Related Issue**
Closes #1381 

- [x]  **Functionality validation**
    - [x]  Feature working as described in the related issue
- [x]  **Product validation**
    - [x]  Preview link if it’s a customer facing issue
- [x] **Documentation**
    - [x] New API changes are documented

### CODE REVIEW

File: src/back/routes/proposal.ts [hunk 1]
```diff
@@ -645,33 +645,25 @@ async function getGrants(): Promise<CategorizedGrants> {
...
 
-  const current = filterGrants(grantsResult.current)
-  const past = filterGrants(grantsResult.past)
-
   return {
-    current,
-    past,
-    total: current.length + past.length,
+    data: filterGrants(grantsResult.data),
+    total: grantsResult.data.length,
```
Comment (ncomerci) [inline src/back/routes/proposal.ts:666]: if you are filtering the grants, I guess the lenght will not be the same

**Review by ncomerci — APPROVED** [review]

File: src/back/routes/proposal.ts [hunk 2]
```diff
@@ -645,33 +645,27 @@ async function getGrants(): Promise<CategorizedGrants> {
...
+  const grantsResult = await ProjectService.getProjects()
 
-  const filterGrants = (grants: ProjectWithUpdate[]) => {
-    return grants.filter(
-      (grant) => grant.user.toLowerCase() === address.toLowerCase() || coauthoringProposalIds.has(grant.id)
+  const filterGrants = (projects: ProjectWithUpdate[]) => {
+    return projects.filter(
+      (project) =>
+        project.type === ProposalType.Grant &&
+        (project.user.toLowerCase() === address.toLowerCase() || coauthoringProposalIds.has(project.id))
```
Comment (1emu) [inline src/back/routes/proposal.ts:660]: You can use `isSameAddress` for this :)

File: src/components/Profile/GrantBeneficiaryList.tsx [hunk 3]
```diff
@@ -1,25 +1,37 @@
...
+import FullWidthButton from '../Common/FullWidthButton'
 
 import GrantBeneficiaryItem from './GrantBeneficiaryItem'
 
 interface Props {
+  address: string | null
   grants: Project[]
 }
 
 const MAX_GRANTS = 4
```
Comment (1emu) [inline src/components/Profile/GrantBeneficiaryList.tsx:15]: 👁️! The issue definition says this should be 3 :)

File: src/back/routes/proposal.ts (hunk 2 above)
Comment (1emu) [reply src/back/routes/proposal.ts:660]: I'm thinking we could solve all this in ProposalModel just by querying:

```sql
-- get the authored grants
SELECT * FROM proposals as p
WHERE p.user = '0x549a9021661a85b6bc51c07b3a451135848d0048'
and p.type = 'grant'

UNION

-- get the coauthored grants
SELECT proposals.*
FROM proposals
    JOIN coauthors
        ON proposals.id = coauthors.proposal_id
WHERE
    coauthors.address = '0x549a9021661a85b6bc51c07b3a451135848d0048' AND
    coauthors.status = 'APPROVED' AND
    proposals.type = 'grant';
```

We could add order by date, and then separate them between past and current

**Review by 1emu — APPROVED** [review]

File: src/back/routes/proposal.ts (hunk 2 above)
Comment (andyesp) [reply src/back/routes/proposal.ts]: Agree! I'll create an issue to do this optimization.

**Pull request merged**